## Inegi Jalisco Census Data

Goal
- Drive Insights with education and economics for each municipality in the Mexico state of Jalisco for a subequent geographic analysis. 

Approach:
1. First look at the pueblos by population strictly and split into quartiles based on population total s
2. Second look at specific municipalies and group them by that 
3. Finally look at specific pueblos and what each one says based on census data (this one will take the longest)

In [2]:
import pandas as pd
import numpy as np 

In [3]:
#load in the data
df = pd.read_csv('RESAGEBURB2020 - 14 Jalisco (1).csv')

- PEA: Population over 12 that is economically active or has job
- PEA_F: Population over 12 that is economically active or has job for females
- PEA_M: Population over 12 that is economically active or has job for males
- PEA_INAC:Population over 12 that is NOT economically active 
- PE_INAC_F:Population over 12 that is NOT economically active for females
- PE_INAC_M:Population over 12 that is NOT economically active for males
- GRAPROES:Median school grade 
- GRAPROES_F: Median school grade for females
- GRAPROES_M: Median school grade for males
- P15YM_SE: Population over 15 without school
- PROM_HNV:Amount of children
- POBTOT: Total population 

In [4]:
df.head(5)

ENTIDAD  NOM_ENT  MUN NOM_MUN  LOC NOM_LOC  AGEB  MZA POCUPADA PEA  ...  \
0       14  Jalisco    1  Acatic    1  Acatic  0194   18       33  33  ...   
1       14  Jalisco    1  Acatic    1  Acatic  0194   19       32  32  ...   
2       14  Jalisco    1  Acatic    1  Acatic  0194    2       23  24  ...   
3       14  Jalisco    1  Acatic    1  Acatic  0194   20       31  31  ...   
4       14  Jalisco    1  Acatic    1  Acatic  0194   21       32  33  ...   

  POCUPADA_M PDESOCUP PDESOCUP_F PDESOCUP_M GRAPROES GRAPROES_F GRAPROES_M  \
0         21        0          0          0     7.76       8.38       7.21   
1         17        0          0          0     8.33       8.48       8.14   
2         18        *          0          *     6.51       5.91       7.14   
3         24        0          0          0     6.90       6.04       7.59   
4         19        *          0          *     6.89       6.47       7.38   

  P15YM_SE PROM_HNV POBTOT  
0        *     1.82     66  
1        8     1.96     61  
2        6     3.26     55  
3        5     3.36     72  
4       11     3.03     61  

[5 rows x 26 columns]

In [53]:
#change dtypes of every column into an int64
print(df.dtypes)

NOM_ENT       object
MUN            int64
NOM_MUN       object
AGEB          object
PEA           object
PEA_F         object
PEA_M         object
PE_INAC       object
PE_INAC_F     object
PE_INAC_M     object
POCUPADA_F    object
POCUPADA_M    object
PDESOCUP      object
PDESOCUP_F    object
PDESOCUP_M    object
GRAPROES      object
GRAPROES_F    object
GRAPROES_M    object
P15YM_SE      object
PROM_HNV      object
POBTOT         int64
dtype: object


In [56]:
columns_to_convert = ['PEA', 'PEA_F', 'PEA_M', 'PE_INAC', 'PE_INAC_F',
       'PE_INAC_M', 'POCUPADA_F', 'POCUPADA_M', 'PDESOCUP', 'PDESOCUP_F',
       'PDESOCUP_M', 'GRAPROES', 'GRAPROES_F', 'GRAPROES_M', 'P15YM_SE',
       'PROM_HNV', 'POBTOT']

In [60]:
#replace all the * with nan so that I can then change the object columns into float columns and work with them more flexibly
df.replace({'*': np.nan, '?': np.nan}, inplace=True)

In [62]:
df.head()

NOM_ENT  MUN NOM_MUN  AGEB PEA PEA_F PEA_M PE_INAC PE_INAC_F PE_INAC_M  \
0  Jalisco    1  Acatic  0194  33    12    21      14        10         4   
1  Jalisco    1  Acatic  0194  32    15    17      16        13         3   
2  Jalisco    1  Acatic  0194  24     5    19      21        18         3   
3  Jalisco    1  Acatic  0194  31     7    24      31        21        10   
4  Jalisco    1  Acatic  0194  33    13    20      23        17         6   

   ... POCUPADA_M PDESOCUP PDESOCUP_F PDESOCUP_M GRAPROES GRAPROES_F  \
0  ...         21        0          0          0     7.76       8.38   
1  ...         17        0          0          0     8.33       8.48   
2  ...         18      NaN          0        NaN     6.51       5.91   
3  ...         24        0          0          0     6.90       6.04   
4  ...         19      NaN          0        NaN     6.89       6.47   

  GRAPROES_M P15YM_SE PROM_HNV POBTOT  
0       7.21      NaN     1.82     66  
1       8.14        8     1.96     61  
2       7.14        6     3.26     55  
3       7.59        5     3.36     72  
4       7.38       11     3.03     61  

[5 rows x 21 columns]

In [61]:
df[columns_to_convert] = df[columns_to_convert].astype('float')

ValueError: could not convert string to float: 'N/D'

In [5]:
df = df[['NOM_ENT', 'MUN', 'NOM_MUN', 'AGEB','PEA', 'PEA_F', 'PEA_M', 'PE_INAC', 'PE_INAC_F',
       'PE_INAC_M', 'POCUPADA_F', 'POCUPADA_M', 'PDESOCUP', 'PDESOCUP_F',
       'PDESOCUP_M', 'GRAPROES', 'GRAPROES_F', 'GRAPROES_M', 'P15YM_SE',
       'PROM_HNV', 'POBTOT']]

In [6]:
df = df.dropna()

In [7]:
df['NOM_MUN'].unique()

array(['Acatic', 'Atemajac de Brizuela', 'Tomatlán', 'Tonalá',
       'Zacoalco de Torres', 'Atenguillo', 'Zapopan', 'Zapotlanejo',
       'San Ignacio Cerro Gordo', 'Atotonilco el Alto', 'Atoyac',
       'Autlán de Navarro', 'Ayotlán', 'Ayutla', 'La Barca', 'Bolaños',
       'Acatlán de Juárez', 'Cabo Corrientes', 'Casimiro Castillo',
       'Cihuatlán', 'Encarnación de Díaz', 'Etzatlán', 'El Grullo',
       'Guachinango', 'Guadalajara', 'Ameca', 'Mexticacán', 'Mezquitic',
       'Mixtlán', 'Ocotlán', 'Ojuelos de Jalisco', 'Pihuamo', 'Poncitlán',
       'El Salto', 'San Cristóbal de la Barranca',
       'San Diego de Alejandría', 'San Juan de los Lagos', 'San Julián',
       'San Marcos', 'San Martín de Bolaños', 'San Martín Hidalgo',
       'Tepatitlán de Morelos', 'Tequila', 'Teuchitlán',
       'Tizapán el Alto', 'Tlajomulco de Zúñiga', 'San Pedro Tlaquepaque',
       'Tolimán'], dtype=object)

In [8]:
#change format options to supress scientic notation
pd.options.display.float_format = '{:.2f}'.format

In [9]:
print(df['POBTOT'].describe())

count     30000.00
mean        449.84
std       17073.64
min           0.00
25%          22.00
50%          59.00
75%         114.00
max     1476491.00
Name: POBTOT, dtype: float64


In [10]:
df['POBTOT'].loc[29996]

15

In [11]:
#0-22,23-59,60-114,115- 1476491
Q1_boolean_search = (df['POBTOT'] < 22)
Q1_Pueblos = df[Q1_boolean_search]
Q1_Pueblos = Q1_Pueblos.dropna()

In [12]:
Q2_boolean = (df['POBTOT'] >= 22) & (df['POBTOT'] <= 59)
Q2_Pueblos = df[Q2_boolean]
Q2_Pueblos.head()

NOM_ENT  MUN NOM_MUN  AGEB PEA PEA_F PEA_M PE_INAC PE_INAC_F PE_INAC_M  \
2   Jalisco    1  Acatic  0194  24     5    19      21        18         3   
5   Jalisco    1  Acatic  0194  14     5     9      14        11         3   
7   Jalisco    1  Acatic  0194  16     5    11      15        13         *   
10  Jalisco    1  Acatic  0194  23     9    14      23        17         6   
11  Jalisco    1  Acatic  0194  21    10    11       9         7         *   

    ... POCUPADA_M PDESOCUP PDESOCUP_F PDESOCUP_M GRAPROES GRAPROES_F  \
2   ...         18        *          0          *     6.51       5.91   
5   ...          9        0          0          0     7.25       7.94   
7   ...         11        0          0          0     7.10          7   
10  ...         14        0          0          0     9.88      10.25   
11  ...         11        0          0          0    10.37      10.06   

   GRAPROES_M P15YM_SE PROM_HNV POBTOT  
2        7.14        6     3.26     55  
5        6.33        6     1.44     34  
7        7.21        4     3.50     39  
10       9.42        3        2     57  
11      10.77        3     2.53     38  

[5 rows x 21 columns]

In [13]:
Q3_boolean = (df['POBTOT']>=60) & (df['POBTOT']<=114)
Q3_boolean

0         True
1         True
2        False
3         True
4         True
         ...  
29995     True
29996    False
29997     True
29998    False
29999     True
Name: POBTOT, Length: 30000, dtype: bool

In [19]:
Q3_Pueblo = df[Q3_boolean]

#Q3_Pueblo['NOM_MUN'].unique()

In [23]:
Q4_boolean = (df['POBTOT']>=115) & (df['POBTOT']<=1476491)
Q4_Pueblo = df[Q4_boolean]

In [48]:
Q1_Pueblos.describe()

MUN  POBTOT
count 7305.00 7305.00
mean    64.91    6.85
std     37.61    7.12
min      1.00    0.00
25%     22.00    0.00
50%     70.00    5.00
75%     98.00   13.00
max    125.00   21.00

In [47]:
Q2_Pueblos.describe()

MUN  POBTOT
count 7751.00 7751.00
mean    71.11   40.18
std     37.50   10.87
min      1.00   22.00
25%     39.00   31.00
50%     74.00   40.00
75%    101.00   50.00
max    125.00   59.00

In [46]:
Q3_Pueblo.describe()

MUN  POBTOT
count 7505.00 7505.00
mean    75.51   84.61
std     36.98   15.69
min      1.00   60.00
25%     39.00   71.00
50%     97.00   84.00
75%    101.00   98.00
max    125.00  114.00

In [50]:
Q4_Pueblo.describe()

MUN     POBTOT
count 7439.00    7439.00
mean    78.61    1680.15
std     35.97   34259.31
min      1.00     115.00
25%     39.00     137.00
50%     97.00     170.00
75%    101.00     258.00
max    125.00 1476491.00

## Observations 
- Q1_Pueblos has 7305 records
- Q2_Pueblos has 7751 records
- Q3_Pueblos has 7505 records 
- Q4_Purblos has 7439 records